components/function-controller/pkg/errors/error.go

```diff
@@ -0,0 +1,51 @@
+package errors
+
+import (
+	"fmt"
+)
+
+type code int8
+
+const (
+	errEmptyValue code = iota + 1
+	errInvalidValue
+)
+
+type xtdError struct {
+	msg  interface{}
+	code code
+}
+
+func (e xtdError) Error() string {
```

Comment:
you could add compile time check for xtdError to satisy Error, like `var _ Error = &xtdError`

Reply:
Or whatever that `error` interface is called in golang std lib